packages/sit-onyx/src/utils/forms.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { expect, test } from "vitest";
+import { getFirstInvalidType } from "./forms";
+
+const getTestState = (overrideData: Partial<ValidityState>): ValidityState => ({
+  valid: false,
+  badInput: false,
+  customError: false,
+  patternMismatch: false,
+  rangeOverflow: false,
+  rangeUnderflow: false,
+  stepMismatch: false,
+  tooLong: false,
+  tooShort: false,
+  typeMismatch: false,
+  valueMissing: false,
+  ...overrideData,
+});
+
+test("should return nothing when there is no invalid state", () => {
+  // ARRANGE
+  const testState = getTestState({ valid: true });
+  // ACT
+  const result = getFirstInvalidType(testState);
+  // ASSERT
+  expect(result).toBeUndefined();
+});
+
+test("should prioritize valueMissing over other invalid types", () => {
+  // ARRANGE
+  const testState = getTestState({ badInput: true, valueMissing: true });
+  // ACT
+  const result = getFirstInvalidType(testState);
+  // ASSERT
+  expect(result).toBe("valueMissing");
+});
+
+test("should return a single invalid type", () => {
+  // ARRANGE
+  const testState = getTestState({ tooShort: true });
+  // ACT
+  const result = getFirstInvalidType(testState);
+  // ASSERT
+  expect(result).toBe("tooShort");
+});
+
+test("should return the first invalid type that is found", () => {
```

Comment:
Imho the test name is a little confusing because "badInput" is returned although its the third key in the object :)
I mean it makes sense because badInput is the first key in the ValidityState object but maybe we can adjust the test name :)

Reply:
I was worried that the descriptions get too long, but it's more important that it is understandable so I changed it ^^
better? :)